deploy/buildtools/buildconfig.py

```diff
@@ -1,177 +1,107 @@
-    def __init__(self, name, buildconfigdict, launch_time):
+    """Represents a single build configuration used to build RTL, drivers, and bitstreams.
+
+    Attributes:
+        name: Name of config i.e. name of `config_build_recipe.yaml` section.
+        build_config_file: Pointer to global build config file.
+        TARGET_PROJECT: Target project to build.
+        DESIGN: Design to build.
+        TARGET_CONFIG: Target config to build.
+        deploytriplet: Deploy triplet override.
+        launch_time: Launch time of the manager.
+        PLATFORM_CONFIG: Platform config to build.
+        s3_bucketname: S3 bucketname for AFI builds.
+        post_build_hook: Post build hook script.
+    """
+    name: str
+    build_config_file: BuildConfigFile
+    TARGET_PROJECT: Optional[str]
+    DESIGN: str
+    TARGET_CONFIG: str
+    deploytriplet: Optional[str]
+    launch_time: str
+    PLATFORM_CONFIG: str
+    s3_bucketname: str
+    post_build_hook: str
+
+    def __init__(self,
+            name: str,
+            recipe_config_dict: Dict[str, Any],
+            build_config_file: BuildConfigFile,
+            launch_time: str) -> None:
+        """
+        Args:
+            name: Name of config i.e. name of `config_build_recipe.yaml` section.
+            recipe_config_dict: `config_build_recipe.yaml` options associated with name.
+            build_config_file: Global build config file.
+            launch_time: Time manager was launched.
+        """
         self.name = name
-        self.TARGET_PROJECT = buildconfigdict.get('TARGET_PROJECT')
-        self.DESIGN = buildconfigdict['DESIGN']
-        self.TARGET_CONFIG = buildconfigdict['TARGET_CONFIG']
-        self.PLATFORM_CONFIG = buildconfigdict['PLATFORM_CONFIG']
-        self.instancetype = buildconfigdict['instancetype']
-        self.deploytriplet = buildconfigdict['deploytriplet']
+        self.build_config_file = build_config_file
+
+        self.TARGET_PROJECT = recipe_config_dict.get('TARGET_PROJECT')
+        self.DESIGN = recipe_config_dict['DESIGN']
+        self.TARGET_CONFIG = recipe_config_dict['TARGET_CONFIG']
+        self.deploytriplet = recipe_config_dict['deploy-triplet']
         self.launch_time = launch_time
-        self.launched_instance_object = None
 
-    def __repr__(self):
-        return "BuildConfig obj:\n" + pprint.pformat(vars(self), indent=10)
+        # run platform specific options
+        self.PLATFORM_CONFIG = recipe_config_dict['PLATFORM_CONFIG']
+        self.s3_bucketname = recipe_config_dict['s3-bucket-name']
+        if valid_aws_configure_creds():
+            aws_resource_names_dict = aws_resource_names()
+            if aws_resource_names_dict['s3bucketname'] is not None:
+                # in tutorial mode, special s3 bucket name
+                self.s3_bucketname = aws_resource_names_dict['s3bucketname']
+        self.post_build_hook = recipe_config_dict['post-build-hook']
```

Comment:
I still don't understand why the s3bucketename has been pushed down into the individual configs instead of being in `BuildConfigFile` but hopefully that will become apparent as we move forward.

Reply:
Answered in more depth in the meeting. I basically am preparing for the `what-to-build` and `where-to-run` PRs (all platform-specific items should be enveloped in the build recipe)